feat: Implement 'uninstall' CLI subcommand for package removal

eyallampel1 opened the pull request:
## Overview
This pull request introduces the 'uninstall' subcommand as per the requirements of Issue #16. It allows users to remove packages using the CLI, streamlining the package management process within the CoCMD environment.

## Changes
- **Uninstallation Logic**: Implemented in `uninstall.rs`, encapsulating the logic required for cleanly removing packages.
- **Packages Management**: Enhanced [packages_manager.rs] to facilitate the removal of packages from both the internal HashMap and the `packages.txt` file.
- **CLI Integration**: Extended `mod.rs` and `main.rs` to incorporate the 'uninstall' subcommand, enabling it to be part of the standard CLI workflow.

## Usage
With this update, users can execute `cocmd uninstall <package_name>` to remove a package. This command ensures that the package is not only removed from the internal tracking but also from the persistent storage, maintaining consistency across the application state.

## Testing
- Manual tests were conducted to ensure that the uninstallation process is smooth and error-free.
- Verified that the `packages.txt` file is updated accordingly post-uninstallation.
- Ensured backward compatibility with existing functionalities.

## Request for Merge
I believe this update fulfills the requirements of Issue #16 and enhances the usability of CoCMD. I have tested the changes and ensured that they meet the project's standards. I kindly request a review for this pull request and suggest any further improvements or merge if it meets the project's expectations.

mzsrtgzr2 commented on src/core/packages_manager.rs:
```diff
@@ -26,18 +26,43 @@ impl PackagesManager {
+        }
     }
 
     fn save_packages(packages_file: &str, packages: &HashMap<String, Package>) {
```
so maybe to remove `save_packages` its not used anymore

mzsrtgzr2 reviewed (changes requested):
remove save_packages

eyallampel1 commented:
I have removed the unnecessary `save_packages` function as suggested. Additionally, I have updated my branch with the latest changes from the upstream master to ensure that there are no merge conflicts. Please review the updated pull request
